plugins/ftdidmx/FtdiWidget-libftdi.cpp

```diff
@@ -47,36 +54,170 @@ using std::vector;
 
 FtdiWidget::FtdiWidget(const string& serial,
                        const string& name,
-                       uint32_t id)
+                       uint32_t id,
+                       const uint16_t vid,
+                       const uint16_t pid
+                      )
     : m_serial(serial),
       m_name(name),
-      m_id(id) {
-  bzero(&m_handle, sizeof(struct ftdi_context));
+      m_id(id),
+      m_vid(vid),
+      m_pid(pid) {}
+
+FtdiWidget::~FtdiWidget() {}
+
+/* @brief
+ * Get the number of physical interfaces our widgit has to offer.
+ *
+ * This does not deal with product names being named in a different way.
+ *
+ * Originally I had hoped to use ftdi_context::type however, it only gets set
+ * properly after the device has been opened.
+ */
+int FtdiWidget::GetInterfaceCount() {
+  std::string tmp = m_name;
+  ToLower(&tmp);
+  if (std::string::npos != tmp.find("plus4")) {
+    return 4;
+  } else  if (std::string::npos != tmp.find("plus2")) {
+    return 2;
+  } else {
+    return 1;
+  }
+}
+
+// @brief Build a list of all attached ftdi devices
```

Comment:
Can you put this back to the other style again please:
/**
 * @brief Build a list...
 */